### DISCUSSION TITLE
Query Params cause App re render

### DISCUSSION THREAD

**Discussion by khuongtp** [vercel/next.js#45033] [Help]
Hi! I have a quick question that why App gets re-rendered when we add a query param like ?test=test. Also, if I use dynamic import, App also gets re-rendered. I tried creating a new project with the latest version, disabled reactStrictMode, put some console.log like this
![image](https://user-images.githubusercontent.com/62872729/213373813-ca45de40-d881-46bb-8888-e6710407c658.png)
![image](https://user-images.githubusercontent.com/62872729/213374136-55a51acb-7ced-4c59-ad3e-0c348cc5c377.png)
and then go to localhost:3000?test=test
![image](https://user-images.githubusercontent.com/62872729/213374248-c1437db0-354c-4a43-94bf-8f5521d95aea.png)
I don't know if it's a bug or not, any idea? Thank you!

**Comment by icyJoseph** [comment]
Hi,

This has to do with how the next router deals with queries. Basically, pages are pre-rendered, and during pre-rendering there's no queries, because it all happens in the server, there's no browser at all.

So once we are in the client, if there's queries, the Next router, immediately does a re-render.

I've answered to this, extensively here: https://github.com/vercel/next.js/discussions/40749#discussioncomment-3700478

The docs talk about it here: https://nextjs.org/docs/advanced-features/automatic-static-optimization#how-it-works:

> During prerendering, the router's query object will be empty since we do not have query information to provide during this phase. After hydration, Next.js will trigger an update to your application to provide the route parameters in the query object.
> 
> The cases where the query will be updated after hydration triggering another render are:
> 
> The page is a [dynamic-route](https://nextjs.org/docs/routing/dynamic-routes).
> The page has query values in the URL.
> [Rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) are configured in your next.config.js since these can have parameters that may need to be parsed and provided in the query.

**Reply by khuongtp** [reply]
Makes sense, thanks for your reply, but one more thing that when I use dynamic import, App also gets re rendered. Do you know why? Thanks again.

**Reply by icyJoseph** [reply]
Could you provide some pseudo code to make sure we are talking about the same thing?

**Reply by khuongtp** [reply]
Let's say I have a Hello component like this
![image](https://user-images.githubusercontent.com/62872729/213382784-8c19cc07-39c9-4eca-b689-ffaf24fe8d96.png)
and then import it dynamically into Home page
![image](https://user-images.githubusercontent.com/62872729/213382932-53ed3e03-53ef-4c4f-b1e5-3abebd3bdc02.png)
Now it renders 3 times
![image](https://user-images.githubusercontent.com/62872729/213383084-92d530f5-995e-44b1-9033-107f328f2646.png)

**Reply by icyJoseph** [reply]
I do not observe the same. With Strict Mode set to false, and a dynamic component, on next@13.1.2, I see two render passes.

![Screenshot 2023-01-19 at 08 58 26](https://user-images.githubusercontent.com/21013447/213386303-a022cbe8-d255-49a7-a7ca-bc3f22aa6d2d.png)

**Reply by khuongtp** [reply]
Could you try hitting reload or empty cache and hard reload a couple times. Sometimes it logs twice, sometimes 3 times

**Reply by icyJoseph** [reply]
Could you prepare a stackblitz repro of what you are seeing? https://stackblitz.com/edit/nextjs-lua5qh?file=README.md
